core/templates/components/state-editor/state-hints-editor/state-hints-editor.component.ts

```diff
@@ -13,231 +13,227 @@
-require('domain/exploration/HintObjectFactory.ts');
-require('domain/utilities/url-interpolation.service.ts');
-require(
-  'components/state-editor/state-editor-properties-services/' +
-  'state-editor.service.ts');
-require(
-  'components/state-editor/state-editor-properties-services/' +
-  'state-hints.service.ts');
-require(
-  'components/state-editor/state-editor-properties-services/' +
-  'state-interaction-id.service.ts');
-require(
-  'components/state-editor/state-editor-properties-services/' +
-  'state-next-content-id-index.service');
-require(
-  'components/state-editor/state-editor-properties-services/' +
-  'state-solution.service.ts');
-require('filters/format-rte-preview.filter.ts');
-require('services/alerts.service.ts');
-require('services/context.service.ts');
-require('services/editability.service.ts');
-require('services/generate-content-id.service.ts');
-require('services/contextual/window-dimensions.service.ts');
-require('services/external-save.service.ts');
-require('services/ngb-modal.service.ts');
-
+import { Component, EventEmitter, OnInit, Output } from '@angular/core';
+import { downgradeComponent } from '@angular/upgrade/static';
+import { Hint } from 'domain/exploration/HintObjectFactory';
+import INTERACTION_SPECS from 'interactions/interaction_specs.json';
+import { UrlInterpolationService } from 'domain/utilities/url-interpolation.service';
+import { StateEditorService } from 'components/state-editor/state-editor-properties-services/state-editor.service';
+import { StateHintsService } from 'components/state-editor/state-editor-properties-services/state-hints.service';
+import { StateInteractionIdService } from 'components/state-editor/state-editor-properties-services/state-interaction-id.service';
+import { StateNextContentIdIndexService } from 'components/state-editor/state-editor-properties-services/state-next-content-id-index.service';
+import { StateSolutionService } from 'components/state-editor/state-editor-properties-services/state-solution.service';
+import { FormatRtePreviewPipe } from 'filters/format-rte-preview.pipe';
+import { AlertsService } from 'services/alerts.service';
+import { EditabilityService } from 'services/editability.service';
+import { WindowDimensionsService } from 'services/contextual/window-dimensions.service';
+import { ExternalSaveService } from 'services/external-save.service';
 import { AddHintModalComponent } from 'pages/exploration-editor-page/editor-tab/templates/modal-templates/add-hint-modal.component';
+import { DeleteHintModalComponent } from 'pages/exploration-editor-page/editor-tab/templates/modal-templates/delete-hint-modal.component';
+import { DeleteLastHintModalComponent } from 'pages/exploration-editor-page/editor-tab/templates/modal-templates/delete-last-hint-modal.component';
+import { NgbModal } from '@ng-bootstrap/ng-bootstrap';
+import { CdkDragSortEvent, moveItemInArray} from '@angular/cdk/drag-drop';
+
+interface deleteResponse {
```

Comment:
Name of interface should start with a capital letter.

Reply:
done